lib/Logger/LogThread.cpp

```diff
@@ -29,7 +29,7 @@
 using namespace arangodb;
 
 LogThread::LogThread(application_features::ApplicationServer& server, std::string const& name)
-    : Thread(server, name), _messages(16) {}
+    : Thread(server, name), _messages(16384) {}
```

Comment:
I am not sure if this change is justified.
The boost::lockfree::queue we are using _can_ already grow dynamically. So I think we can get away with a much smaller reserve size, e.g. 32 or 64, and then have it grow dynamically if it actually needs more space (it will never do in most environments).
```suggestion
    : Thread(server, name), _messages(64) {}
```

Reply:
If that is actually the case, then I think we need a new theory why a `LOG_TOPIC` is blocking the thread issuing this log message for a long time, when the disk is slow and busy. @neunhoef

Reply:
I understand that @neunhoef has measured a case, where LOG_TOPIC took 1.4 seconds to finish.

Reply:
The boost::lockfree::queue is an implementation of the Michael-Scott queue, which is based on a linked list and is effectively unbounded. However, the boost::lockfree::queue uses a freelist to manage the available nodes (and thereby work around the memory reclamation problem), and by limiting the available nodes on the free list, it can effectively limit the number of elements in the queue, making it bounded/fixed size. But for that the queue has to be defined using the `boost::lockfree::fixed_sized<true>` policy, which is not the case here. The number passed to the constructor only defines the number of preallocated nodes on the freelist. For a non-fixed sized queue like this, new nodes will be allocated on demand if the freelist is empty, so I agree with @jsteemann here.

Reply:
We can go to 64, no problem. But we still need to understand how a LOG_TOPIC operation can take 1.4 seconds in situations with a bad IO system...

Reply:
Done.

Reply:
@neunhoef can you share some details under which circumstances you observe this behavior? I'm wondering if it might be possible to reproduce this locally so I could investigate.